modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -359,37 +384,55 @@ function ActionsDropdownRenderer({actions, itemData, itemId}) {
 		return <ClayLoadingIndicator small />;
 	}
 
-	const renderItems = (items) =>
-		items.map(({items: nestedItems = [], separator, type, ...item}, i) => {
-			if (type === 'group') {
-				return (
-					<ClayDropDown.Group {...item}>
-						{separator && <ClayDropDown.Divider />}
+	const renderItems = ({actions, quickActionItems = false}) =>
+		actions.map(
+			({items: nestedItems = [], separator, type, ...item}, i) => {
+				if (type === 'group') {
+					return (
+						<ClayDropDown.Group {...item}>
+							{separator && <ClayDropDown.Divider />}
+
+							{renderItems({
+								actions: nestedItems,
+								quickActionItems,
+							})}
+						</ClayDropDown.Group>
+					);
+				}
 
-						{renderItems(nestedItems)}
-					</ClayDropDown.Group>
+				return (
+					<ActionItem
+						action={item}
+						closeMenu={() => setMenuActive(false)}
+						handleAction={handleAction}
+						itemData={itemData}
+						itemId={itemId}
+						key={i}
+						method={item.method ?? item.data?.method}
+						quickActionItem={quickActionItems}
+						setLoading={setLoading}
+						url={item.href && formatActionURL(item.href, itemData)}
+					/>
 				);
 			}
-
-			return (
-				<ActionItem
-					action={item}
-					closeMenu={() => setMenuActive(false)}
-					handleAction={handleAction}
-					itemData={itemData}
-					itemId={itemId}
-					key={i}
-					method={item.method ?? item.data?.method}
-					setLoading={setLoading}
-					url={item.href && formatActionURL(item.href, itemData)}
-				/>
-			);
-		});
+		);
 
 	return (
 		<div className="d-flex justify-content-end ml-auto">
 			{inlineEditingAlwaysOn && inlineEditingActions}
 
+			{quickActionsEnabled && formattedActions.length > 1 && (
+				<div className="quick-action-menu">
```

Comment:
Minor nit: can we call this class `quick-actions-menu`? A menu holds multiple actions :slightly_smiling_face:

Reply:
Hmm I'm afraid this branding comes [from Clay](https://github.com/liferay/clay/blob/aa4653f901d22386711b2807c501e9e94a2a1dd5/packages/clay-css/src/scss/components/_quick-action.scss)
The renaming would break the styles we use from there... 😢

Reply:
Ok, it's best to leave it as it is then.